src/index.ts

```diff
@@ -1,8 +1,27 @@
 import realm from 'realm'
+import sift from 'sift'
 
 const aggregate = (aggregate: any[], realm: realm, model: string ) => {
+	return aggregate.reduce((prevVal, curVal, index, array) => {
+		return applyAggregation(array[index], prevVal);
+	},realm.objects(model).toJSON())
+}
+
+const applyAggregation = (aggregation: any, data: any) => {
+	for( const stage in aggregation){
+		const query = aggregation[stage]
+		switch(stage){
+			case '$match': return applyMatchStage(query, data)
+			default: {
+				console.warn("stage not implemented: ", stage)
+				return data
```

Comment:
@takameyer How about throwing an error here instead of returning `data`? Because I am afraid having `data` returned in the default case might cause bugs that are not easily traceable.

Reply:
I'm fine with that.